lib/ipam/ipam.go

```diff
@@ -661,82 +667,21 @@ func (c ipamClient) ReleaseIPs(ctx context.Context, ips []net.IP) ([]net.IP, err
 	return unallocated, nil
 }
 
-func (c ipamClient) releaseIPsFromBlock(ctx context.Context, ips []net.IP, blockCIDR net.IPNet) ([]net.IP, error) {
+func (c ipamClient) releaseIPsFromBlock(ctx context.Context, ips []net.IP, blockCIDR net.IPNet, node v3.Node) ([]net.IP, error) {
 	logCtx := log.WithField("cidr", blockCIDR)
 	for i := 0; i < ipamEtcdRetries; i++ {
 		logCtx.Info("Getting block so we can release IPs")
-		obj, err := c.client.Get(ctx, model.BlockKey{CIDR: blockCIDR}, "")
+		obj, err := c.getBlockAndReleaseIfNotSelectsNode(ctx, node, blockCIDR)
```

Comment:
I'd rather not combine so much into one function here - let's keep the original Get() here and just encapsulate the "maybeReleaseAffinity()" part. WDYT?